Data/NumSequenceGenerator/NumSequenceGenerator.cpp

```diff
@@ -0,0 +1,34 @@
+#include "NumSequenceGenerator.h"
+
+NumSequenceGenerator::NumSequenceGenerator()
+{
+	CyclePathFinder path_finder;
+	path_finder.init_numpad();
+	path_finder.find_paths(generate_rand_number(0, 8));
+	num_sequence_ = get_random_int_path(path_finder.get_paths());
+}
+
+std::vector<int> NumSequenceGenerator::get_num_sequence()
+{
+	return num_sequence_;
+}
+
+
+int NumSequenceGenerator::generate_rand_number(const int& min, const int& max)
+{
+		srand(time(nullptr));
+	
+	return min + rand() % ((max + 1) - min);
+}
+
+std::vector<int> NumSequenceGenerator::get_random_int_path(const std::vector<std::vector<Node>>& paths)
+{
+	std::vector<Node> path = paths[generate_rand_number(0, 8)];
+	std::vector<int> seq;
+
+	for (auto node : path)
+		seq.push_back(node.value);
```

Comment:
braces around code blocks are preferred as this approach is less error-prone (e.g. you'll want to add another line of code inside of the loop, then it will be easy to forget that the line will be actually outside of the loop unless you add braces, and nobody will ever notice. It's a common logical error, which developers spend hours to find)

```suggestion
	for (auto node : path) {
		seq.push_back(node.value);
	}
```

Reply:
Added brackets everywhere in the project.